packages/apputils-extension/src/notificationplugin.tsx

```diff
@@ -641,19 +643,41 @@ namespace Private {
    */
   let toastify: typeof ReactToastify | null = null;
 
-  function CloseButton(props: CloseButtonProps): JSX.Element {
-    const trans = translator.load('jupyterlab');
+  export interface ICloseButtonProps
+    extends React.DetailedHTMLProps<
+      React.ButtonHTMLAttributes<HTMLButtonElement>,
+      HTMLButtonElement
+    > {
+    close: (e?: React.MouseEvent<HTMLElement, MouseEvent>) => void;
+    closeIcon: LabIcon.IReact;
+    closeIconMargin?: boolean;
```

Comment:
if you could just add doc string for those to ease code maintenance, it will be great.

Other than that the PR looks good.

Reply:
Thank you, added the doc strings

Reply:
@fcollonval Would you say that docstrings should be added to everything public? So in this case because public element uses interface, adding docstrings to interface is enough. If there would be no interface, we would want to have docstrings on the public element, right?